Classes/Controller/FormController.php

```diff
@@ -21,12 +21,9 @@ class FormController extends \In2code\Powermail\Controller\FormController
     public function formAction(): ResponseInterface
     {
         if ($this->settings['powermailcaptcha']['useSiteLanguage'] ?? 0) {
-            $currentLanguageId = GeneralUtility::makeInstance(Context::class)
-                ->getPropertyFromAspect('language', 'id');
-            $siteLanguage = $this->request->getAttribute('site')->getLanguageById($currentLanguageId);
-            $this->view->assign('languageIso', $siteLanguage->getTwoLetterIsoCode());
+            $this->view->assign('languageIso', $GLOBALS['TYPO3_REQUEST']->getAttribute('language')->getTwoLetterIsoCode());
```

Comment:
Since this is an Extbase action, you might be able to do this to get the language:

`$language = $this->request->getAttribute('language');`

see

* request object: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/Typo3Request.html
*  fetching language from request: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/Language.html

It is not really recommended to use $GLOBALS['LANG'], $GLOBALS['TSFE'],  $GLOBALS['TYPO3_REQUEST'] etc. anymore unless absolutely necessary and we put quite some effort into updating the docs on docs.typo3.org, adding cross-links etc.

A full list of all the GLOBALS with links to more in-depth info is on https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/Configuration/GlobalVariables.html (though I just realized TYPO3_REQUEST is missing there, will create PR, but we recently added TSFE and LANG).

Reply:
Correction: above change not possible with TYPO3 10 support, sorry. 

With powermail ^8.4 support you have support for TYPO3 ^10.4, so you can't make suggested change in that case as it is not available until v11. https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/Typo3Request.html#extbase-controller

Sorry, I missed that, I thought I remembered that the extension supported v11 and up.

Reply:
my code works with TYPO3 v10, I implemented it there. don't mix up `$this->request` with `$GLOBALS['TYPO3_REQUEST']`

Reply:
my code works with TYPO3 v10, I implemented it there. don't mix up `$this->request` with `$GLOBALS['TYPO3_REQUEST']`